Library/Homebrew/resource.rb

```diff
@@ -78,8 +78,11 @@ def owner=(owner)
   end
 
   def downloader
-    @downloader ||= download_strategy.new(url, download_name, version,
-                                          mirrors: mirrors.dup, **specs)
+    return @downloader if @downloader.present?
+
+    urls = determine_mirror_urls
```

Comment:
```suggestion
    url, mirrors = determine_mirror_urls
```

would be a nicer API